### PR TITLE
Fix narrow python

### PR DESCRIPTION
**Pull request by jvanasco**
This is a lightweight attempt to handle the "narrow Python build" I mentioned in #122

If a ValueError is detected, it looks in the exception message (Py2 and Py3 compliant) to see if the 'narrow Python build' warning is detected.  If so, it emits the message as a `warning` and returns the label unaltered.

I only put the docstring on `decode_punycode` to pass tests.

### CODE REVIEW

File: tests/main_test.py [hunk 1]
```diff
@@ -216,3 +216,19 @@ def test_result_as_dict():
                      'domain': 'google',
                      'suffix': 'com'}
     assert result._asdict() == expected_dict
+
+
+def test_punycode_extended():
+    """
+    "narrow python" builds may break on certain unicode characters
```
Comment (john-kurkowski) [inline tests/main_test.py]: Can you colocate this test with other punycode test(s)? Can you expand this comment how it differs from those other tests? 

CI passes with or without the library code change in this PR. Worth mentioning that? And what the test _should_ do on a narrow build? I'm fine with "should." For this edge case, I don't care to maintain a narrow build environment in CI. :blush:

File: tldextract/tldextract.py [hunk 2]
```diff
@@ -227,11 +229,19 @@ def __call__(self, url):
         labels = netloc.split(".")
 
         def decode_punycode(label):
+            """helper function"""
```
Comment (john-kurkowski) [inline tldextract/tldextract.py]: While you're in here, could be a little more specific?

File: CHANGELOG.md [hunk 3]
```diff
@@ -3,6 +3,12 @@
 After upgrading, update your cache file by deleting it or via `tldextract
 --update`.
 
+## 2.1.0 (2017-05-24)
```
Comment (john-kurkowski) [inline CHANGELOG.md]: Accidental commit slip in here?

Comment (jvanasco) [inline CHANGELOG.md]: I'll try to rebase again before fixing the other issues; might an issue with this version of git. If so, I'll try from another machine.  Github keeps thinking the existing merge already in master is new.

**Comment by jvanasco** [comment]
I largely addressed the changes.  I'll repatch against a new master if they're okay - that should get past the git issues.

I'm possibly doing a bad job explaining the tests/passing, so I'll try rephrasing it a bit here...

If one has a "narrow" Python build (which is the default for python through 3.3), then the punycode decoding will raise an exception on a UTF-16 character.  This fix just leaves it as punycode - which is what happens if a unicode error is encountered.  I don't think it would be likely to submit an encoded UTF-16 into the function - though it may be possible.  (the domain that triggers this, incidentally, is a url shortener that turns links into emojis - because of course)

Tests could easily be set to only run on narrow builds... I thought it would make sense to run in all environments

    import sys
    import pytest


    def possible_narrow_unicode():
        """
        Python3 doesn't have a "narrow" unicode problem.
        Python2 may either be built with a narrow unicode, or have expanded support
        """
        if sys.version_info < (3, 0):
            if sys.maxunicode <= 65535:
                return True
        return False


    SKIP_TESTS_NARROW_UNICODE = pytest.mark.skipif(not possible_narrow_unicode(),
                                                   reason="not a narrow unicode build")

    @SKIP_TESTS_NARROW_UNICODE
    def test_punycode_narrow_python():
        ....

**Comment by john-kurkowski** [comment]
That sample `skipif` code is nice. I'd almost prefer it, to document the test better: the test only applies in situation X. Unfortunately, I don't think the test suite exercises situation X. Not without further changes to tox.ini and .travis.yml. So, better a test that always runs, vs. a test that never runs and is forgotten.

Your expanded test docstring is what I was looking for. Can it also clarify the new test case is redundant on a wide build, that it is primarily to cover a narrow build? Again, your `skipif` code would've showed this rather than tell it. But no need for further maintenance hoops.

File: CHANGELOG.md [hunk 4]
```diff
@@ -3,6 +3,17 @@
 After upgrading, update your cache file by deleting it or via `tldextract
 --update`.
 
+## 2.1.1 (?)
```
Comment (john-kurkowski) [inline CHANGELOG.md]: No need to fill out release details in this PR. I can take care of that when I release.

**Comment by john-kurkowski** [comment]
> I'll repatch against a new master if they're okay - that should get past the git issues.

Rebasing atop the latest upstream branch, Git would've seen your original 0eeafb5e224f77938e11d9a2a754c9f22ee7bd26 was an empty commit and omitted it. Or interactively, stopped to suggest omitting it. Now that this PR contains an actual CHANGELOG.md diff, rebasing will update the diff to contain just that actual, new change. No duplicate diff with `master`. (However, [I've asked here](https://github.com/john-kurkowski/tldextract/pull/130#discussion_r118845892), can you delete the commit entirely?)

**Comment by jvanasco** [comment]
I would have preferred for the skipif, and decided against it, for the same reasons as you. I looked into supporting a narrow Python build and it just didn't seem worth setting up a custom Python binary.  I have narrow and wide builds on my boxes, most people don't.

I'll give the rebase another try - it just wasn't coming out cleanly .  If it doesn't work, I diff'd this branch into a new one with a proper, clean, single-commit. I'll replace this branch with that one if I can't git to play nice, as I don't have time to upgrade it (every box here has an older git)

**Comment by hangtwenty** [comment]
@jvanasco do you mind resolving the conflict? thanks!
